src/pip/_internal/req/req_install.py

```diff
@@ -866,6 +874,7 @@ def install(
             )
         except LegacyInstallFailure as exc:
             self.install_succeeded = False
+            assert exc.__cause__ is not None
             raise exc.__cause__
```

Comment:
```suggestion
            if exc.__cause__ is not None:
                raise exc.__cause__
            raise
```

Maybe?

Reply:
Or maybe we should add an attribute on `LegacyInstallFailure` instead of depending on `__cause__`.